tests/api.c

```diff
@@ -41261,6 +41261,297 @@ static int test_wolfSSL_X509_bad_altname(void)
+        0x78, 0x61, 0x6d, 0x70, 0x6c, 0x65, 0x31, 0x32, 0x37, 0x2e, 0x63, 0x6f,
+        0x6d, 0x82, 0x0e, 0x65, 0x78, 0x61, 0x6d, 0x70, 0x6c, 0x65, 0x31, 0x32,
+        0x38, 0x2e, 0x63, 0x6f, 0x6d, 0x82, 0x0e, 0x65, 0x78, 0x61, 0x6d, 0x70,
+        0x6c, 0x65, 0x31, 0x32, 0x39, 0x2e, 0x63, 0x6f, 0x6d, 0x82, 0x0e, 0x65,
+        0x78, 0x61, 0x6d, 0x70, 0x6c, 0x65, 0x31, 0x33, 0x30, 0x2e, 0x63, 0x6f,
+        0x6d, 0x30, 0x1d, 0x06, 0x03, 0x55, 0x1d, 0x0e, 0x04, 0x16, 0x04, 0x14,
+        0xae, 0xe3, 0xd3, 0x4e, 0x2c, 0x04, 0x01, 0xa7, 0xb1, 0xdc, 0x6d, 0xe4,
+        0x93, 0xb0, 0x79, 0xf5, 0xdc, 0x73, 0xd8, 0x94, 0x30, 0x0d, 0x06, 0x09,
+        0x2a, 0x86, 0x48, 0x86, 0xf7, 0x0d, 0x01, 0x01, 0x0b, 0x05, 0x00, 0x03,
+        0x82, 0x01, 0x01, 0x00, 0x8f, 0xd3, 0x2b, 0x5d, 0x4b, 0x74, 0x7c, 0x4f,
+        0x37, 0x73, 0x55, 0x92, 0x12, 0xf0, 0x7f, 0xb7, 0x33, 0x50, 0xae, 0x95,
+        0xaa, 0x23, 0x10, 0xb4, 0xe7, 0xa3, 0xde, 0xd4, 0x51, 0x19, 0xc1, 0xcb,
+        0x98, 0x72, 0xad, 0x7e, 0x3b, 0x65, 0x26, 0xe2, 0x82, 0x44, 0xf4, 0xde,
+        0x1d, 0x7f, 0x24, 0xa5, 0xbb, 0x83, 0xe5, 0xc1, 0xcf, 0xce, 0x3f, 0x0a,
+        0x56, 0xf5, 0x01, 0xb6, 0x04, 0x59, 0x8e, 0xd0, 0x36, 0x79, 0xdc, 0xde,
+        0x4e, 0x9d, 0x63, 0x1e, 0xc0, 0xe7, 0xa5, 0xe8, 0x89, 0x80, 0x6f, 0x7a,
+        0xac, 0x94, 0x92, 0x96, 0x7e, 0x14, 0xe1, 0x81, 0xaa, 0x7d, 0x61, 0xc4,
+        0xe5, 0xee, 0x93, 0x09, 0x02, 0xe5, 0xf0, 0xc1, 0x57, 0x20, 0xd8, 0x81,
+        0x96, 0xb8, 0xe4, 0x16, 0x21, 0x69, 0xd8, 0x9a, 0x08, 0x5b, 0xf5, 0x33,
+        0xd1, 0x88, 0x69, 0x2a, 0xf1, 0xac, 0xd9, 0x86, 0xcc, 0xab, 0xc9, 0xc4,
+        0xbb, 0xb8, 0x61, 0x5e, 0x72, 0xee, 0x49, 0xb5, 0x23, 0xa0, 0xc9, 0xf1,
+        0x03, 0x44, 0x72, 0xc0, 0xac, 0x4c, 0xea, 0xaf, 0xea, 0x2a, 0x20, 0x1f,
+        0x44, 0xff, 0x9e, 0x9d, 0x4f, 0x18, 0xff, 0x83, 0x48, 0x53, 0xc3, 0x91,
+        0xff, 0xee, 0xd6, 0x56, 0xa7, 0x49, 0x65, 0xf3, 0x07, 0x26, 0x9f, 0x9c,
+        0x42, 0x23, 0x12, 0xd9, 0xc0, 0x91, 0x16, 0x82, 0x60, 0x14, 0x2e, 0xbc,
+        0xf0, 0xad, 0x65, 0xdf, 0xa0, 0xf2, 0x20, 0xbe, 0xe4, 0xc4, 0x00, 0xf2,
+        0x31, 0xbd, 0x81, 0x80, 0xc7, 0xc3, 0xd5, 0x54, 0x75, 0x39, 0x34, 0xad,
+        0x55, 0xb4, 0x0c, 0xd2, 0xb7, 0x8b, 0xe3, 0x7e, 0x5b, 0x1d, 0xb3, 0x0c,
+        0x3c, 0xdb, 0x36, 0x8b, 0x2d, 0xe1, 0xd3, 0x0a, 0x55, 0xc0, 0x60, 0x50,
+        0xc8, 0xd4, 0x1f, 0x46, 0x6a, 0x90, 0xee, 0x0f, 0x57, 0x83, 0x96, 0x2d,
+        0xff, 0x9f, 0xea, 0x78, 0x6f, 0x11, 0x9d, 0xe6
+    };
+
+    X509* x509 = NULL;
+    int certSize = (int)sizeof(too_many_altnames_cert) / sizeof(unsigned char);
+
+    ExpectNull(x509 = wolfSSL_X509_load_certificate_buffer(
+        too_many_altnames_cert, certSize, SSL_FILETYPE_ASN1));
+#endif
+#endif
+    return EXPECT_RESULT();
+}
+
+static int test_wolfSSL_X509_max_name_constraints(void)
+{
+    EXPECT_DECLS;
+#if !defined(NO_FILESYSTEM) && !defined(NO_CERTS) && !defined(NO_RSA) && \
+    !defined(NO_WOLFSSL_CLIENT)
```

Comment:
I'm surprised `NO_WOLFSSL_CLIENT` is a condition.

Reply:
Since I was using wolfSSL_CTX_load_verify_locations_ex() I modeled the flag guards the same as its test function test_wolfSSL_CTX_load_verify_locations_ex(). I think its needed because I need wolfSSLv23_client_method().

Reply:
Fair. You could use `wolfSSLv23_server_method()` in that case. And gate on `NO_WOLFSSL_CLIENT`. Not really a big deal.

Reply:
I considered that as well actually, do you think one is better than the other?

Reply:
You can use either. Just use the following to cover both cases:

```
#ifndef NO_WOLFSSL_SERVER
        ExpectNotNull(ctx = wolfSSL_CTX_new(wolfSSLv23_server_method()));
#else
        ExpectNotNull(ctx = wolfSSL_CTX_new(wolfSSLv23_client_method()));
#endif
```

Reply:
New guards are in commit.